Library/Homebrew/cmd/vendor-install.sh

```diff
@@ -119,7 +119,7 @@ fetch() {
     --remote-time
     --location
     --user-agent "${HOMEBREW_USER_AGENT_CURL}"
-    --header "Authorization: Bearer ${HOMEBREW_DOCKER_REGISTRY_TOKEN:-QQ==}"
+    --header "Authorization: ${HOMEBREW_GITHUB_PACKAGES_AUTH}"
```

Comment:
Why not just expose `HOMEBREW_GITHUB_PACKAGES_AUTH` so that those who need a basic token can set
```
HOMEBREW_GITHUB_PACKAGES_AUTH="Basic ..."
```
?

This way we don't meet additional special handling of various variables.

Reply:
If I was doing this again from scratch: sure, but we'd need to deprecate the existing variables and I'm not sure it's worth the effort at this point.

Reply:
Well, I was thinking we could keep the existing variables and use `HOMEBREW_GITHUB_PACKAGES_AUTH` as an override to everything else rather than introduce a new `HOMEBREW_DOCKER_REGISTRY_BASIC_AUTH_TOKEN` and keep track of `HOMEBREW_GITHUB_PACKAGES_AUTH` privately. But this is fine too.

Reply:
No new configuration variables introduced in this PR.

Reply:
Ah, yes, I see.